### ISSUE TITLE
How to stop ultralytics from setting default running folder

### ISSUE THREAD

**Issue by JasonSloan** [ultralytics/ultralytics#18193]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, team!
Like the title says, how to stop ultralytics from setting default running folder?
Technically, i like the way in yolov5 when training or testing a model, it will create the "runs" folder inside the project. 
However, in the newer version of yolo like yolov8 and yolov11, the program always creates the "runs" folder somewhere else except inside the project. 
How can i stop this? Does there exist a parameter that can be passed to control whether i can make the program set by itself

Best wishes!

**Comment by pderrenger** [comment]
👋 Hello @JasonSloan, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for the latest usage information. For example, the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) sections might help clarify how configuration settings work, as well as address other frequently asked questions.

For your query, it seems related to a feature change in newer YOLO versions regarding the default folder structure. An Ultralytics engineer will review this shortly and provide precise guidance. 

In the meantime:
- If you're looking to modify the output folder location, you might want to inspect your specific configuration file or relevant function calls.
- For additional reproducibility during testing, we recommend sharing relevant code snippets and specifying the exact versions of `ultralytics`, `Python`, and `PyTorch` you are using. This helps in diagnosing the issue efficiently.

If this is more of a ❓ Question rather than reporting a bug, kindly provide as much context as possible, including any parameters or configurations you’ve tried, to help us understand your problem better.

## Upgrade

Ensure you're using the latest `ultralytics` package for the best experience, as new updates often address reported issues or introduce useful features. Update using:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following environments with dependencies pre-installed:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Join the Ultralytics community 🌟! Share insights or ask questions on [Discord](https://discord.com/invite/ultralytics) 🎧, the [community forum](https://community.ultralytics.com), or [Reddit](https://reddit.com/r/Ultralytics). An Ultralytics engineer will get back to you shortly! 😊

**Comment by Laughing-q** [comment]
@JasonSloan Hey can you paste the output of `yolo settings` here? and also please provide a minimum reproduce command or script to reproduce your issue. Thanks

**Comment by JasonSloan** [comment]
> @JasonSloan Hey can you paste the output of `yolo settings` here? and also please provide a minimum reproduce command or script to reproduce your issue. Thanks

Hi, thank you for your reply!
I didnt use the cli directly, instead, i downloaded the source code from the repository and run inside it, namely, created a python file named 'train.py' and wrote a few lines of codes to call ultralytics API.

**Comment by Y-T-G** [comment]
> I didnt use the cli directly, instead, i downloaded the source code from the repository and run inside it, namely, created a python file named 'train.py' and wrote a few lines of codes to call ultralytics API.

You can still install and should install a local clone by running `pip install -e .` inside the directory.

**Comment by JasonSloan** [comment]
> > I didnt use the cli directly, instead, i downloaded the source code from the repository and run inside it, namely, created a python file named 'train.py' and wrote a few lines of codes to call ultralytics API.
> 
> You can still install and should install a local clone by running `pip install -e .` inside the directory.

Hi, there wasnt any running problem. 
I just want to make the code create the "runs" folder(which includes the training info) inside my current-cloned-source-code-project rather than somewhere else. 
And i dont like using the cli, I just wrote a few lines of codes below to call, so that i can debug into the source code easily.


`from ultralytics import YOLO

weight = "weights/yolo11s.pt"
model = YOLO(weight)
model.train(
    data='ultralytics/cfg/datasets/wheelhouse.yaml',
    cfg='ultralytics/cfg/default.yaml',
    name='exp',
    epochs=200,
    device=[0, 1],
    workers=8,
    optimizer='Adam',
    lr0=1e-4,
    patience=50,
    multi_scale=True,
    label_smoothing=True
)`

**Comment by JasonSloan** [comment]
Hi there, I found the solution.
I need to specify project='.', so that the code can create the 'runs' folder inside the current project.
So is there a list of legal varibles that can be customrized and passed into the function and thus i can reference?

Thanks!

**Comment by glenn-jocher** [comment]
Glad you found the solution! You can reference all customizable variables in the [Ultralytics Settings Docs](https://docs.ultralytics.com/quickstart/#ultralytics-settings) or directly inspect them via `print(settings)` in Python. For task-specific arguments, check the [YOLO Python API Guide](https://docs.ultralytics.com/quickstart/). Let us know if you need further clarification!

**Comment by Y-T-G** [comment]
You can also just use `yolo settings runs_dir="./runs"` to change the settings to use current directory for runs.

**Comment by JasonSloan** [comment]
Got it! Thanks a lot!

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask. Happy experimenting with YOLO!
